### PR TITLE
feat: List class

### PR DESCRIPTION
**Pull request by RogerMo01**

### CODE REVIEW

File: OOP-Exercises/MyList.cs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OOP_Exercises
+{
+    class MyList
+    {
+        public int[] Sequence { get; set; }
+        public int Count => Sequence.Length;
+
```
Comment (LongJohnSilver1504) [inline OOP-Exercises/MyList.cs:13]: a lot of empty spaces here

File: OOP-Exercises/MyList.cs [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+        /// <summary>
+        /// Initialize an empty MyList
+        /// </summary>
+        public MyList() => Sequence = new int[0];
+        /// <summary>
```
Comment (LongJohnSilver1504) [inline OOP-Exercises/MyList.cs:19]: Leave 1 empty space between methods

File: OOP-Exercises/MyList.cs [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+        /// <summary>
+        /// Adds an int element to the end of MyList
+        /// </summary>
+        /// <param name="element"></param>
+        public void Add(int element)
+        {
+            int[] newSequence = new int[Sequence.Length + 1];
+
+            Array.Copy(Sequence, newSequence, Sequence.Length);
```
Comment (LongJohnSilver1504) [inline OOP-Exercises/MyList.cs:34]: Why not avoid array copy after every add.... we can do this every 100 or 1000 adds

File: OOP-Exercises/MyList.cs (hunk 1 above)
Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:13]: Fixed!

File: OOP-Exercises/MyList.cs (hunk 2 above)
Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:19]: OK!

Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:19]: Done!

File: OOP-Exercises/MyList.cs (hunk 3 above)
Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:34]: OK!

Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:34]: Done!

File: OOP-Exercises/MyList.cs [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+            Count = baseArray.Length;
+        }
+
+        /// <summary>
+        /// Adds an int element to the end of MyList
+        /// </summary>
+        /// <param name="element"></param>
+        public void Add(int element)
+        {
+            if (Count == 1000)
```
Comment (LongJohnSilver1504) [inline OOP-Exercises/MyList.cs:36]: Here we have the same problem but now when we pass the 1000 elements.... we copy the array  every time

File: OOP-Exercises/MyList.cs [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        /// <summary>
+        /// Searches the entire sorted MyList for an element using the default comparer.
+        /// Return: The index of the int value in MyList, if value is found; otherwise, a negative number.
+        /// If this method is called with a non-sorted MyList, the return value can be incorrect and a negative number could be returned, even if value is present in MyList.
+        /// </summary>
+        /// <param name="element"></param>
+        /// <returns></returns>
+        public int BinarySearch(int element)
+        {
+            return Array.BinarySearch(Sequence, element);
```
Comment (LongJohnSilver1504) [inline OOP-Exercises/MyList.cs:74]: LOL! 🐀  Do your own binary search!

Comment (RogerMo01) [reply OOP-Exercises/MyList.cs:74]: Ok. Easy

Comment (RogerMo01) [reply OOP-Exercises/MyList.cs]: Done! What do you think??

File: OOP-Exercises/MyList.cs (hunk 4 above)
Comment (RogerMo01) [reply OOP-Exercises/MyList.cs]: Done!

**Review by LongJohnSilver1504 — APPROVED** [review]
LGTM!

**Pull request merged**